akka-stream/src/main/scala/akka/stream/impl/io/FileSubscriber.scala

```diff
@@ -66,13 +70,13 @@ private[akka] class FileSubscriber(f: File, bytesWrittenPromise: Promise[Long],
         chan.force(true)
       } catch {
         case ex: Exception ⇒
-          bytesWrittenPromise.failure(ex)
+          completionPromise.success(IOResult(bytesWritten, Failure(ex)))
       }
       context.stop(self)
   }
 
   override def postStop(): Unit = {
-    bytesWrittenPromise.trySuccess(bytesWritten)
+    completionPromise.trySuccess(IOResult(bytesWritten, Success(Done)))
```

Comment:
I see. But then I should not only move the completion down one line, but also catch any exceptions from `close()` and complete the promise with `IOResult(bytesWritte, Failure(ex))` if exception is thrown, right?